ortb/clone.go

```diff
@@ -189,6 +189,138 @@ func CloneUser(s *openrtb2.User) *openrtb2.User {
 	return &c
 }
 
+func CloneDevice(s *openrtb2.Device) *openrtb2.Device {
+	if s == nil {
+		return nil
+	}
+
+	// Shallow Copy (Value Fields)
+	c := *s
+
+	// Deep Copy (Pointers)
+	c.Geo = CloneGeo(s.Geo)
+
+	c.DNT = CloneInt8Pointer(s.DNT)
+	c.Lmt = CloneInt8Pointer(s.Lmt)
+
+	c.SUA = CloneUserAgent(s.SUA)
+	if s.ConnectionType != nil {
+		connectionTypeCopy := s.ConnectionType.Val()
+		c.ConnectionType = &connectionTypeCopy
+	}
+
+	c.Ext = sliceutil.Clone(s.Ext)
+
+	return &c
+}
+
+func CloneInt8Pointer(s *int8) *int8 {
+	if s == nil {
+		return nil
+	}
+	var dntCopy int8
+	dntCopy = *s
+	return &dntCopy
+}
+
+func CloneUserAgent(s *openrtb2.UserAgent) *openrtb2.UserAgent {
```

Comment:
I don't see a unit test for this function. It appears to be an oversight as I see all other functions have their own dedicated unit test.

Reply:
Added, good catch! Sorry I don't know why I didn't add this test.